README.md

```diff
@@ -29,19 +30,79 @@ Or simply use pip or pip3
 pip3 install cerberus-python-client
 ```
 
+Alternatively, add `cerberus-python-client` in the `install_requires` section of your project's `setup.py`.
+Then run one of the following from within your projects directory:
+```bash
+python3 setup.py install
+```
+or for python 2.7
+```bash
+python setup.py install
+```
+
 ## Usage
 
+#### Import the Client:
+
 ```python
 from cerberus.client import CerberusClient
 ```
-This client supports 2 different types of authentication, both of which returns a Vault Token.
 
-* username and password (CLI usage)
+#### Instantiate the Client
+
+
+Default IAM Role Authentication:
+
+```python
+client = CerberusClient('https://my.cerberus.url')
+```
+
+Assumed Role Authentication:
+```python
+client = CerberusClient('https://my.cerberus.url', role_arn='arn:aws:iam::0000000000:role/role-name')
+```
+
+User Authentication:
 ```python
 client = CerberusClient('https://my.cerberus.url', username, password)
 ```
 
-* EC2 IAM role or Lambda(default mode)
+#### Read Secrets from Cerberus
+
+To get a secret for a specific key in a safe deposit box:
+```python
+client.get_secret("app/path/to/secret", "secretName")
+```
+
+To get all the secrets for an safe deposit box:
+```python
+client.get_secrets("app/path/to/secret")
+```
+
+#### Get a Cerberus Authentication token
+
+If you do not want to read a secret, but simply want an authentication token, then you can use one of the `<type>_auth.py` classes to retrieve a token.
+
+You can also use the CerberusClient class.
+
+* IAM Role Authentication
+```python
+from cerberus.aws_auth import AWSAuth
+token = AWSAuth('https://my.cerberus.url').get_token()
+```
+
+* Assumed Role Authentication
```

Comment:
What does this mean?  Is the Python client going to try to assume that role via an AWS call?  Or is it going to just send that role to AWS?

If it is the latter, shouldn't it be "IAM Role Authentication with statically supplied role"?